### DISCUSSION TITLE
Apply functions to a vector auxvariable

### DISCUSSION THREAD

**Discussion by Minjiang-Zhu** [idaholab/moose#22752] [Q&A General] [answered]
I have two scalar functions and I want to apply them to the components of a vector auxvariable, what should I do? Please notice the functions are not simply functions of x, y, z, and t (they are read from the solution file of the previous step and then are modified a bit).

**Comment by Minjiang-Zhu** [comment]
I know there is an auxkernal called 'VectorFunctionAux' which can assign a vector-function to a vector auxvariable. If someone knows how to combine two scalar functions into a vector function, that will also help.

**Comment by GiudGiud** [comment, marked as answer]
Hello

VectorFunctionAux will let you set the auxiliary variable using a Function that has a vectorValue() routine. It does NOT parse postprocessor names or scalar variables directly. A [MooseParsedVectorFunction](https://mooseframework.inl.gov/source/functions/MooseParsedVectorFunction.html#) can do that upstream of the auxkernel. 

Guillaume
